src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala

```diff
@@ -312,8 +328,61 @@ class HudiOutputWriter(props: Map[String, Object], hudiOutput: Option[Hudi]) ext
     df
   }
 
+  def hiveSyncManually(dataFrame: DataFrame, path: Option[String], hudiOutput: Option[Hudi]): DataFrame = {
+    var df = dataFrame
+    val ss = df.sparkSession
+    val catalog = ss.catalog
+
+    df = getHudiDf(df, ss, path)
+
+    val tablesToSync = getTablesToSyncByStorageType(hudiOutput, hudiOutputProperties.tableName.get)
+
+    for ((table, tableInputFormat) <- tablesToSync) {
+      val identifier = new TableIdentifier(table, Option(catalog.currentDatabase))
+      val tableType = CatalogTableType("EXTERNAL")
+
+      val storage = new CatalogStorageFormat(Option(new URI(path.get)),
+        Option(tableInputFormat),
+        Option("org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat"),
+        Option("org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe"),
+        false,
+        Map[String, String]())
+
+      val tableDefinition = new CatalogTable(identifier, tableType, storage, df.schema)
+      ss.sharedState.externalCatalog.createTable(tableDefinition, true)
+      ss.sharedState.externalCatalog.alterTableDataSchema(catalog.currentDatabase, table, df.schema)
+      // Handle partitions
+      //ss.sharedState.externalCatalog.createPartitions(ss.catalog.currentDatabase, table,
+        //Seq(CatalogTablePartition(hudiOutputProperties.partitionExpressionManual.get, storage)), true)
+
+    }
+    return df
+  }
+
 
-   }
+  def getHudiDf(dataFrame: DataFrame, ss: SparkSession, path: Option[String]): DataFrame = {
+
+    var df = dataFrame
+
+    val hudiPath: String = (hudiOutputProperties.partitionExpressionManual) match {
+      case (Some(partitionExpressionManual)) => path.get + "/*/*.parquet"
+      case (None) => path.get
+    }
+    df = ss.read.format("org.apache.hudi").option("path", hudiPath).load()
+    hudiOutputProperties.partitionExpressionManual.get.foreach(partition => df.drop(partition._1))
```

Comment:
Do it the other way around:
```df.drop(Get an array from partitionExpressionManual```
You also need to surround with case, since you're not always going to have partitionExpressionManual